src/features/projectsV2/ProjectsMap/SingleProjectView.tsx

```diff
@@ -54,15 +63,52 @@ const SingleProjectView = ({ mapRef }: { mapRef: MapRef }) => {
         sitesGeojson,
         selectedSite,
         setViewState,
-        4000
+        3500
       );
     }
   }, [selectedSite, requestedSite, router.isReady]);
+
+  useEffect(() => {
+    if (
+      singleProject &&
+      hasNoSites &&
+      !selectedPlantLocation &&
+      router.isReady
+    ) {
+      const latitude = singleProject.coordinates.lat;
+      const longitude = singleProject.coordinates.lon;
+      // Zoom into location for a project  which has no site
+      zoomToLocation(setViewState, longitude, latitude, 10, 3500, mapRef);
+    } else if (
+      selectedPlantLocation?.geometry?.type === 'Point' &&
+      selectedPlantLocation?.geometry.coordinates.length === 2
+    ) {
+      const [lon, lat] = selectedPlantLocation.geometry.coordinates;
+      if (lon !== undefined && lat !== undefined) {
+        // Zoom into single tree registration
+        zoomToLocation(setViewState, lon, lat, 20, 3500, mapRef);
+      }
```

Comment:
Would it be better to separate the code for handling plant location zoom? i.e. group the code that handles zoom into point and polygon plant locations together.

Reply:
Have covered in this (https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/2255) PR